### ISSUE TITLE
Reproduction discrepancy for Carbon Dimer (C2) Triplet state

### ISSUE THREAD

**Issue by ooyaitsuki** [deepqmc/deepqmc#235]
Hello,

I am attempting to reproduce the Carbon Dimer ($C_2$) Triplet state calculations presented in your paper: *"An Improved Penalty-Based Excited-State Variational Monte Carlo Approach with Deep-Learning Ansatzes"* (DOI: 10.1021/acs.jctc.4c00678).

I ran the calculation using `deepqmc` with the `psiformer` ansatz targeting the Triplet state (`spin=2`), but I am observing a systematic discrepancy of approximately **3~6 mHartree** compared to the reference values in the paper.

**1. Comparison of Results:**

| State | My Result (Hartree) | Paper Result (Hartree) | Difference (mHa) |
| :--- | :--- | :--- | :--- |
| State 0 | -75.9074(5) | -75.9132(7) | ~5.8 |
| State 1 | -75.9063(5) | -75.9128(3) | ~6.5 |
| State 2 | -75.8737(5) | -75.8772(1) | ~3.5 |
| State 3 | -75.8666(6) | -75.8733(1) | ~6.7 |

**2. My Configuration:**

I used the following full `config.yaml`. I explicitly set `spin: 2` (Triplet) and the unit to `angstrom`.

<details>
<summary>Click to expand full config.yaml</summary>

```yaml
task:
  workdir: ???
  sampler_factory:
    _target_: deepqmc.sampling.initialize_sampling
    _partial_: true
    elec_sampler:
      _target_: deepqmc.sampling.combine_samplers
      _partial_: true
      samplers:
      - _target_: deepqmc.sampling.DecorrSampler
        length: 10
      - _target_: deepqmc.sampling.LangevinSampler
        _partial_: true
        tau: 1.0
  opt:
    _target_: deepqmc.optimizer.KFACOptimizer
    _partial_: true
    kfac:
      _target_: kfac_jax.Optimizer
      _partial_: true
      learning_rate_schedule:
        _target_: deepqmc.utils.InverseSchedule
        init_value: 0.05
        decay_rate: 10000
      norm_constraint: 0.001
      damping_schedule:
        _target_: deepqmc.utils.ConstantSchedule
        value: 0.001
      estimation_mode: fisher_exact
      num_burnin_steps: 0
      inverse_update_period: 1
  _target_: deepqmc.app.train_from_factories
  hamil: ${hamil}
  ansatz: ${ansatz}
  steps: 100000
  molecule_batch_size: 1
  electron_batch_size: 2048
  electronic_states: 4
  seed: 0
  pretrain_steps: 1000
  pretrain_kwargs:
    opt: adam
    opt_kwargs:
      learning_rate: 0.0003
      b1: 0.9
      b2: 0.999
    scf_kwargs:
      basis: aug-cc-pVTZ
      cas:
      - 8
      - 8
  loss_function_factory:
    _target_: deepqmc.loss.create_loss_fn
    _partial_: true
    alpha: 4.0
    clip_mask_fn:
      _target_: deepqmc.loss.median_log_squeeze_and_mask
      _partial_: true
    clip_mask_overlap_fn:
      _target_: deepqmc.loss.psi_ratio_clip_and_mask
      _partial_: true
  mols:
    _target_: deepqmc.app.read_molecules
    directory: null
  metric_logger_constructor:
    _target_: deepqmc.log.TensorboardMetricLogger
    _partial_: true
    max_queue: 10
  chkpt_constructor:
    _target_: deepqmc.log.CheckpointStore
    _partial_: true
  h5_logger_constructor:
    _target_: deepqmc.log.H5Logger
    _partial_: true
logging:
  deepqmc: 10
  jax: 40
  kfac: 40
ansatz:
  _target_: deepqmc.wf.NeuralNetworkWaveFunction
  _partial_: true
  envelope:
    _target_: deepqmc.wf.env.ExponentialEnvelopes
    _partial_: true
    isotropic: true
    per_shell: false
    per_orbital_exponent: true
    spin_restricted: false
    init_to_ones: true
    softplus_zeta: false
  backflow_op:
    _target_: deepqmc.wf.nn_wave_function.BackflowOp
    _partial_: true
    mult_act: '${eval:"lambda x: x"}'
  n_determinants: 16
  full_determinant: true
  cusp_electrons:
    _target_: deepqmc.wf.cusp.ElectronicCuspAsymptotic
    _partial_: true
    same_scale: 0.25
    anti_scale: 0.5
    alpha: 1.0
    trainable_alpha: true
    cusp_function:
      _target_: deepqmc.wf.cusp.PsiformerCusp
  cusp_nuclei: false
  backflow_transform: mult
  conf_coeff:
    _target_: deepqmc.hkext.SumPool
    _partial_: true
  omni_factory:
    _target_: deepqmc.wf.omni.OmniNet
    _partial_: true
    embedding_dim: 256
    jastrow_factory: null
    backflow_factory:
      _target_: deepqmc.wf.omni.Backflow
      _partial_: true
      subnet_factory:
        _target_: deepqmc.hkext.MLP
        _partial_: true
        hidden_layers:
        - log
        - 1
        bias: false
        last_linear: true
        activation: null
        init: ferminet
    nuclear_gnn_head: null
    gnn_factory:
      _target_: deepqmc.gnn.ElectronGNN
      _partial_: true
      n_interactions: 4
      nuclei_embedding: null
      electron_embedding:
        _target_: deepqmc.gnn.electron_gnn.ElectronEmbedding
        _partial_: true
        positional_embeddings:
          ne:
            _target_: deepqmc.gnn.edge_features.CombinedEdgeFeature
            features:
            - _target_: deepqmc.gnn.edge_features.DistancePowerEdgeFeature
              powers:
              - 1
              log_rescale: true
            - _target_: deepqmc.gnn.edge_features.DifferenceEdgeFeature
              log_rescale: true
        use_spin: true
        project_to_embedding_dim: true
      two_particle_stream_dim: 32
      self_interaction: true
      edge_features: null
      layer_factory:
        _target_: deepqmc.gnn.electron_gnn.ElectronGNNLayer
        _partial_: true
        subnet_factory:
          _target_: deepqmc.hkext.Identity
          _partial_: true
        electron_residual: false
        nucleus_residual: false
        two_particle_residual: false
        deep_features: false
        update_rule: concatenate
        update_features:
        - _target_: deepqmc.gnn.update_features.NodeAttentionElectronUpdateFeature
          _partial_: true
          num_heads: 4
          mlp_factory:
            _target_: deepqmc.hkext.MLP
            _partial_: true
            hidden_layers:
            - log
            - 2
            bias: true
            last_linear: false
            activation:
              _target_: jax.numpy.tanh
              _partial_: true
            init: ferminet
          attention_residual:
            _target_: deepqmc.hkext.ResidualConnection
            normalize: false
          mlp_residual:
            _target_: deepqmc.hkext.ResidualConnection
            normalize: false
hamil:
  _target_: deepqmc.hamil.MolecularHamiltonian
  mol:
    _target_: deepqmc.molecule.Molecule
    coords:
    - - -0.622
      - 0.0
      - 0.0
    - - 0.622
      - 0.0
      - 0.0
    charges:
    - 6
    - 6
    charge: 0
    spin: 2
    unit: angstrom
  laplacian_factory:
    _target_: deepqmc.physics.laplacian
    _partial_: true
```
</details>

**3. Questions:**

I suspect that I might not be configuring the `psiformer` ansatz correctly, or that I am missing specific settings required to reproduce the paper's accuracy.

**Could you please share the full `config.yaml` file (or the specific Psiformer hyperparameters) used to obtain the Carbon Dimer results in the paper?**

Having the exact configuration would allow me to directly identify what I am missing in my setup.

Thank you for your help.

**Environment:**
deepqmc 1.2.0
NVIDIA-SMI 570.133.07             Driver Version: 570.133.07     CUDA Version: 12.8
NVIDIA GeForce RTX 4090 x2

**Comment by szbernat** [comment]
Hi,
Thanks for trying deepqmc! Extra thanks for using the hydra command line interface, it makes our job much easier!

For the experiments in this paper, we have created the following task config. Please copy this into a new file `src/deepqmc/conf/task/train_excited_psiformer.yaml`:

```yaml
defaults:
  - sampler: decorr_metropolis_psiformer
_target_: deepqmc.app.train_from_factories
opt:
  _target_: deepqmc.optimizer.KFACOptimizer
  _partial_: true
  kfac:
    _target_: kfac_jax.Optimizer
    _partial_: true
    learning_rate_schedule:
      _target_: deepqmc.utils.InverseSchedule
      init_value: 0.05
      decay_rate: 20000
    norm_constraint: 0.001
    damping_schedule:
      _target_: deepqmc.utils.ConstantSchedule
      value: 0.001
    estimation_mode: fisher_exact
    num_burnin_steps: 0
    inverse_update_period: 1
hamil: ${hamil}
ansatz: ${ansatz}
steps: 50000
molecule_batch_size: 1
electron_batch_size: 2048
electronic_states: ???
pretrain_steps: 5000
pretrain_kwargs:
  opt: lamb
  opt_kwargs:
    learning_rate: 3.e-4
    b1: 0.9
    b2: 0.999
  scf_kwargs:
    basis: ???
    cas: ???
seed: 0
fit_kwargs:
  eval_spin: 100
  loss_function_factory:
    _target_: deepqmc.loss.create_loss_fn
    _partial_: true
    alpha: 4.0
    scale_overlap_by: max_gap_std
    clip_mask_fn:
      _target_: deepqmc.loss.median_clip_and_mask
      _partial_: true
      clip_width: 5
      median_center: true
    clip_mask_overlap_fn:
      _target_: deepqmc.loss.psi_ratio_clip_and_mask
      _partial_: true
metric_logger:
  _target_: deepqmc.log.TensorboardMetricLogger
  _partial_: true
  max_queue: 10
h5_logger:
  _target_: deepqmc.log.H5Logger
  _partial_: true
  keys_whitelist:
    - 'E_loc'
    - 'mol_idxs'
    - 'energy/ewm'
    - 'E_loc_clipped/mean'
    - 'E_loc_clipped/std'
    - 'spin_log'
    - 'overlap/pairwise'
    - 'time'
    - 'step'
mols:
  _target_: deepqmc.app.read_molecules
  directory: null
```
Then you can launch the training using the following command, which adds a few more config options, specific to the carbon dimer. Note that you have to supply your own molecule configuration at `hamil.mol.file=...`
```bash
deepqmc hydra.run.dir=carbon_dimer hamil/mol=from_file hamil.mol.file=<path/to/your/c2/file> +task.pretrain_kwargs.scf_kwargs.fix_spin=2 task.pretrain_kwargs.scf_kwargs.basis=cc-pvdz task.pretrain_kwargs.scf_kwargs.cas=[9,8] task.steps=100000 ansatz=psiformer task=train_excited_psiformer task.electronic_states=4
```
The most important differences from your config seems to me the longer learning rate decay, the different sampler, and the longer pretraining.

Let me know how it goes with this config.

**Comment by zenoone** [comment]
I am closing this as solved. However, if you have any further questions about reproducing the C2 triplet states, please reopen the issue.
